pkg/flow/adapter/xmltojsontransformation/adapter_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xmltojsontransformation
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cetest "github.com/cloudevents/sdk-go/v2/client/test"
+	"github.com/stretchr/testify/assert"
+
+	"knative.dev/eventing/pkg/adapter/v2"
+	logtesting "knative.dev/pkg/logging/testing"
+)
+
+const (
+	tBridgeID         = "bride-abdc-0123"
+	tCloudEventID     = "ce-abcd-0123"
+	tCloudEventType   = "ce.test.type"
+	tCloudEventSource = "ce.test.source"
+	tSuccessAttribute = "success"
+	tErrorAttribute   = "error"
+
+	tXML1        = `<note><to>Tove</to></note>`
+	tJSONOutput1 = "{\"note\": {\"to\": \"Tove\"}}\n"
+
+	tFalseXML         = `"this is not xml"`
+	tFalseXMLResponse = `{"Code":"request-validation","Description":"invalid XML","Details":null}`
+)
+
+func TestXSLTTransformEvents(t *testing.T) {
+	testCases := map[string]struct {
+		inEvent cloudevents.Event
+
+		expectPanic string
+		expectEvent cloudevents.Event
+	}{
+		"transform ok": {
+			inEvent: newCloudEvent(tXML1, cloudevents.ApplicationXML),
+
+			expectEvent: newCloudEvent(tJSONOutput1, cloudevents.ApplicationJSON),
+		},
+		"transform error": {
+			inEvent: newCloudEvent(tFalseXML, cloudevents.ApplicationXML),
+
+			expectEvent: newCloudEvent(tFalseXMLResponse, cloudevents.ApplicationXML),
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+
+			defer func() {
+				r := recover()
+				switch {
+				case r == nil:
+					assert.Empty(t, tc.expectPanic, "Expected panic did not occur")
+				case tc.expectPanic == "":
+					assert.Fail(t, "Unexpected panic", r)
+				default:
+					assert.Contains(t, r, tc.expectPanic)
+				}
+			}()
+
+			ctx := context.Background()
+			logtesting.TestContextWithLogger(t)
```

Comment:
Either this line can be removed, or the result should be assigned to `ctx` instead of the line above, otherwise you're not doing anything with the logger.